bundles/org.openhab.binding.pentair/src/test/java/org/openhab/binding/pentair/internal/PentairParserTest.java

```diff
@@ -0,0 +1,207 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pentair.internal;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.*;
+import static org.openhab.binding.pentair.internal.TestUtilities.parsehex;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.openhab.binding.pentair.internal.parser.PentairIntelliChlorPacket;
+import org.openhab.binding.pentair.internal.parser.PentairParser;
+import org.openhab.binding.pentair.internal.parser.PentairParser.CallbackPentairParser;
+import org.openhab.binding.pentair.internal.parser.PentairStandardPacket;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * PentairParserTest
+ *
+ * @author Jeff James - Initial contribution
+ *
+ */
+@ExtendWith(MockitoExtension.class)
+@NonNullByDefault
+class PentairParserTest {
+    private final Logger logger = LoggerFactory.getLogger(PentairParserTest.class);
+
+    //@formatter:off
+    public static byte[] stream = parsehex(
+            "FF 00 FF A5 1E 0F 10 02 1D 09 1F 00 00 00 00 00 00 00 20 03 00 00 04 3F 3F 00 00 41 3C 00 00 07 00 00 6A B6 00 0D 03 7F"
+                    + "FF 00 FF A5 10 0F 10 12 29 02 E3 02 AF 02 EE 02 BC 00 00 00 02 00 00 00 2A 00 04 00 5C 06 05 18 01 90 00 00 00 96 14 00 51 00 00 65 20 3C 01 00 00 00 07 50 "
+                    + "FF 00 FF A5 01 0F 10 02 1D 0D 1D 20 00 00 00 00 00 00 00 33 00 00 04 4D 4D 00 00 51 6D 00 00 07 00 00 5E D5 00 0D 04 04");
+    //@formatter:on
+
+    PentairParser parser = new PentairParser();
+
+    @Mock
+    @NonNullByDefault({})
+    CallbackPentairParser callback;
+
+    @Captor
+    @NonNullByDefault({})
+    ArgumentCaptor<PentairStandardPacket> packetsStandard;
+
+    @Captor
+    @NonNullByDefault({})
+    ArgumentCaptor<PentairIntelliChlorPacket> packetsIntellichlor;
+
+    @NonNullByDefault({})
+    Thread thread;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        parser.setCallback(callback);
+    }
+
+    @AfterEach
+    public void tearDown() throws Exception {
+        if (thread != null) {
+            thread.interrupt();
+            thread.join();
+        }
+        thread = null;
+    }
+
+    @Test
+    public void test() throws InterruptedException {
+        java.lang.System.setProperty("org.slf4j.simpleLogger.defaultLogLevel", "DEBUG");
```

Comment:
Is this some debugging leftover, i guess this is not something you want to have enabled by default?